### PR TITLE
Getting Started with Polynomial regression in Python

### LINKED ISSUE #4419: [Machine Learning] Getting started with Polynomial regression in python
#### Pre-submission advice
By following all our pre-submission advice and reviewing our Resources folder, you will maximise your chances of your topic being approved.

We ask that you please be patient as our team works through approving and publishing all articles/tutorials in a timely manner.

Allow 1-3 days for a topic to be reviewed and/or approved - allow 3-7 days for an articles to be reviewed and/or published.

Be sure to visit our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for tools, resources, and example articles that will help you propose and write a successful article.

**Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**

**We tend to stray away or tend not to publish reviews/comparisons of commercial product offerings.**

#### Getting started with polynomial regression in python
Your title should be descriptive of the article/tutorial. Be Specific. 

Use keyword research to improve your article's SEO.

#### Proposed article introduction
 Polynomial Regression is a form of linear regression in which the relationship between the study variable x and explanatory variable y is modeled as an nth degree polynomial. This algorithm, it fits a nonlinear relationship between the values of x and the corresponding conditional mean of y, denoted E(y |x). This, model is thus suitable for modeling non-linear relationship between two variables and in such situations linear gression can't fit as the suitable modelling technique.  Polynomial regression is used in the problems such as, salary prediction, pandemic spread prediction, and many others areas of study.

In our article, we shall learn what's the polynomial regression, and precisely walk through the intuition behind this algorithm. We shall then proceed and implement our algorithm using a business datau in python. Finally, we will use our model and predictions using new data points.
#### Key takeaways 
By the end of the article, reader will know:
- What is the polynomial regression
- Situation under which polynomial regression is the suitable model for modelling our data.
- Intuition behind the polynomial regression upon which the we build our algorithm.
- How to implement the polynomial regression model in python.
- How to use and make predictions using the implemented  model  in python.
#### Article quality
How is your article unique? Tell us what makes your approach different from similar articles that have published on the same topic?
Is yours more in-depth? Does it cover additional topics? Do you provide handy tips or anecdotal advice?

#### References
N/A
### Conclusion
Finally, remove the Pre-Submission advice section and all our blockquoted notes as you fill in the form before you submit.
We look forwarding to reviewing your topic suggestion.

### Templates to use as guides
- [How To Guide Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/how-to-guide-template.md)
- [Software Review Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/software-review-template.md)
- [Tutorial Template](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/tutorial-template.md)

### PR DESCRIPTION
**Pull request by Faith034**
### Article checklist

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #4419

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

File: content/articles/polynomial-regression-in-python/index.md [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  - [Step 6: Visualising the Polynomial Regression results](#step-6-visualising-the-polynomial-regression-results)
+  - [Step 7: Visualising the Polynomial Regression results (for higher resolution and smoother curve)](#step-7-visualising-the-polynomial-regression-results-for-higher-resolution-and-smoother-curve)
+  - [Step 8: Predicting a new result with Linear Regression](#step-8-predicting-a-new-result-with-linear-regression)
+  - [Step 9: Predicting a new result with Polynomial Regression](#step-9-predicting-a-new-result-with-polynomial-regression)
+- [Conclussion](#conclussion)
+  
+  # Getting Started with Polynomial Regression in Python
+Polynomial regression is a machine learning model used to model non-linear relationships between dependent and independent variables. Examples of cases where polynomial regression can be used include modelling population growth, the spread of diseases, and epidemics.
+
+## Prerequisites
```
Comment (lalith1403) [inline content/articles/polynomial-regression-in-python/index.md:22]: All headings should be H3 and Beyond.

File: content/articles/polynomial-regression-in-python/index.md [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+where:
+- $M$ is the order of the polynomial.
+- $x^j$ denotes x raised to the power of j.
+- $\bold{w}=w_0 , . . . , w_M$ denotes a vector of weights.
+  
+As we can note from the above, the polynomial function $y(x, \bold w)$ is a non-linear function of $x$. However, this function is a linear function of the weights $\bold w$. Functions that are non-linear in the input variable but linear in the unknown set of the parameters are linear and fall under the *linear models* class. Thus a *polynomial regression* is a **linear model**.
+
+Now the task is to determine the value of $\bold w$ and $M$.
+
+**Obtaining the $\bold w$**
```
Comment (lalith1403) [inline content/articles/polynomial-regression-in-python/index.md:47]: Treat this as a subsection. Use h4 heading.

File: content/articles/polynomial-regression-in-python/index.md [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+```
+
+
+ ```bash
+array([158862.45265155])
+
+ ```
+Executing the code above returns a predicted salary as 158862.45265155. This value lies within the range of our expectations, and thus we can conclude the polynomial regression is suitable for this problem.
+
+ # Conclussion
```
Comment (lalith1403) [inline content/articles/polynomial-regression-in-python/index.md:223]: H3 heading

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Welcome to the EngEd program. You have started your journey on a positive note. The article seems to be really interesting. Can you refer to the contribution guidelines for submission details. Additionally, we will need a short bio and headshot pic for your profile. Refer to the contribution guidelines for adding the required information.

**Comment by Kayere** [comment]
Upon running your article through our 3rd party plagiarism checker it seemed to raise a few flags and the % was higher than we typically accept. Please see attached PDF - and revisit the article to ensure we are contributing wholly unique and original content. Be sure to see our resources page to see more info on plagiarism and what is considered as such.

[Getting Started with Polynomial regression in Python #4694 Copyleaks Report.pdf](https://github.com/section-engineering-education/engineering-education/files/7454487/Getting.Started.with.Polynomial.regression.in.Python.4694.Copyleaks.Report.pdf)

**Comment by lalith1403** [comment]
Have the necessary changes been made? Ping us once done.

**Comment by lalith1403** [comment]
@Faith034 Any updates?

**Comment by mbici** [comment]
@lalith1403 am working on the plagiarism part right now. I'll be notifying you shortly.

**Comment by Faith034** [comment]
@lalith1403
 Am done with the requested changes.

**Comment by lalith1403** [comment]
This article is ready for final review. Can you add the author details (headshot pic and bio) in the authors folder? Thank you.

**Comment by lalith1403** [comment]
Also, a picture is required.

**Comment by Faith034** [comment]
@lalith1403 
Everything is in place now!

File: content/articles/polynomial-regression-in-python/index.md [hunk 4]
```diff
@@ -0,0 +1,280 @@
...
+  
+From the polynomial hypothesis above, we note that this function \(y(x, \bold w)\) is a non-linear function of \(x\). However, this function is a linear function of the weights \(\bold w\). Therefore, functions that are non-linear in the input variable but linear in the unknown set of the parameters are linear and fall under the *linear models* class. Thus a *polynomial regression* is a **linear model**.
+
+Now the task is to determine the value of $\bold w$ and $M$.
+
+#### Obtaining the $\bold w$
+
+To determine the values of the coefﬁcients, we first ﬁt a polynomial to the training dataset. After that, we find the error function, i.e., $E(\bold w)$, which measures the misﬁt between the fitted curve $y(x, \bold w)$ and the data points of the training set. One of the most used error functions in machine learning is the sum of squares of the errors between the predictions $y(x_n, \bold w)$  and the corresponding target values for each point $x_n$ $y_n$ fin the data. This error function is of the form:
+
+\[E(\bold w)=\frac{1}{2}\sum ( {y(x_n , w) − y_n } )^2\]
```
Comment (lalith1403) [inline content/articles/polynomial-regression-in-python/index.md:72]: Wrap these equations within \$\$ to render them

**Review by lalith1403 — CHANGES_REQUESTED** [review]
Review the equations rendering once. The equations need to be wrapped in dollar signs to display the equations.

**Comment by Faith034** [comment]
@lalith1403 
I've enclosed equations within dollar sign and are now okay.

**Comment by hectorkambow** [comment]
jumping on this one

**Pull request merged**